### ISSUE TITLE
Latest arm64 Docker Image is broken

### ISSUE THREAD

**Issue by hutec** [maptiler/tileserver-gl#1031]
When trying to get the latest Docker image (`v4.6.2`) running with
```bash
docker run --rm -it -v $(pwd):/data -p 8080:80 maptiler/tileserver-gl
```

it crashes with the following error:
```error
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package '/usr/src/app/node_modules/request/' imported from /usr/src/app/src/main.js
    at new NodeError (node:internal/errors:405:5)
    at legacyMainResolve (node:internal/modules/esm/resolve:233:9)
    at packageResolve (node:internal/modules/esm/resolve:874:14)
    at moduleResolve (node:internal/modules/esm/resolve:936:20)
    at defaultResolve (node:internal/modules/esm/resolve:1129:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v18.17.1
```

**Comment by hutec** [comment]
Ah just realized that this might be related to #1006?

**Comment by acalcutt** [comment]
This shouldn't be related to #1006 , that has not been implemented yet. request should still exists, it is strange it doesn't.

**Comment by acalcutt** [comment]
I tested it here and it works for me. 

```
root@mapgen3:/opt/wdb-map-gen/tools# docker run --rm -it -v $(pwd):/data -p 8080:8080 maptiler/tileserver-gl
Starting tileserver-gl v4.6.2
No input file specified, using zurich_switzerland.mbtiles
[INFO] Automatically creating config file for zurich_switzerland.mbtiles
[INFO] Only a basic preview style will be used.
[INFO] See documentation to learn how to create config.json file.
Run with --verbose to see the config file here.
Starting server
Listening at http://[::]:8080/
Startup complete
```

maybe run with the version specified to make sure it is using the latest components?
`docker run --rm -it -v $(pwd):/data -p 8080:8080 maptiler/tileserver-gl:v4.6.2`

one thing to note, I see you have 8080:80 in your command, which is out of date. the internal docker port is now 8080 also, so you should have 8080:8080 in you command, as noted in https://maptiler-tileserver.readthedocs.io/en/latest/installation.html#docker

**Comment by hutec** [comment]
Thanks for checking. :)
I tried on two other machines. On another Mac I get the same issue, on a Linux machine it worked.
Strangely, when I build the Docker image locally, it also works on the original machine.
I will investigate a bit more and see whether this is a general issue or a problem with my local setups.

Slightly related, during debugging I saw a deprectation warning for the Node install script in the [CI logs]( https://github.com/maptiler/tileserver-gl/actions/runs/6591264317/job/17909516963#step:16:1528).

**Comment by acalcutt** [comment]
Are the macs failing arm64? maybe the arm docker image failed

**Comment by hutec** [comment]
Yes, indeed, both are `arm64`

**Comment by acalcutt** [comment]
looks like maybe it had a network error building the arm64 version
![Screenshot_20231024_032649_Firefox](https://github.com/maptiler/tileserver-gl/assets/3792408/02270b19-6d3a-4903-956c-e6ba710d6d1a)

**Comment by hutec** [comment]
However, this error wasn't only happening in the last build, but also some of the previous ones.
https://github.com/nautobot/nautobot/pull/4479 sounds like a similar thing

**Comment by MichielMortier** [comment]
I have the same issue with arm64 on my mac

**Comment by MichielMortier** [comment]
Btw, version 4.4.10 still works as a current workaround

**Comment by acalcutt** [comment]
I tested using 'npm install' instead of 'npm ci' but that doesn't seem to have helped. I also tried moving to an older release of ubuntu docker image but that didn't seem to help either....

**Comment by acalcutt** [comment]
Can you try v4.6.3 and tell me if it works for you. I tried it here on a raspberry pi and it seemed to work as expected.

I tried a bunch of things to fix this, but I think in the end what did it was adding "--maxsockets 1" to the npm ci command. It seemed without that, the arm64 builder kept failing on random components. Perhaps npm was making to many connections for this emulated arm64 environment.

**Comment by hutec** [comment]
Works now! Thank you very much for looking into it.

**Comment by acalcutt** [comment]
This re-emerged with the last release v4.6.4 ...so I guess I haven't found a solution yet.

**Comment by acalcutt** [comment]
I made v4.6.5 which seems to have worked with all releases.  unfortunately I'm not sure I have found a real solution. maxsockets seems to help but I'm not sure if it is just a intermittent network issue caused by QEMU or slowness from emulation.  I tried several ideas to fix the ECONNRESET errors on random packages, but not much seemed to help.

I'm going to close this for now, but I think it is going to pop up again on future releases since it seems to happen at random.
